src/stream.js

```diff
@@ -0,0 +1,36 @@
+const { createConnection } = require('ilp-protocol-stream')
+const debug = require('debug')('ilp-fetch:stream')
+const fetch = require('node-fetch')
+
+async function streamPayment ({
+  url,
+  opts,
+  payParams,
+  plugin,
+  payToken
+}) {
+  const [ destinationAccount, _sharedSecret ] = payParams
+  debug('streaming via STREAM. destination=' + destinationAccount)
+
+  const sharedSecret = Buffer.from(_sharedSecret, 'base64')
+  const connection = await createConnection({
+    plugin,
+    destinationAccount,
+    sharedSecret
+  })  
+
+  const stream = connection.createStream()
+  stream.setSendMax(opts.maxPrice)
+
+  await new Promise(resolve => stream.on('data', resolve))
```

Comment:
Why do you need this acknowledgement?

Reply:
Right now there's no way in the STREAM API to detect whether both sides know that the stream is open. So on this side, `isOpen` will return true but on the other side it won't have fired the `on('stream')` event

Reply:
But if the other side sends a manual acknowledgement then you can know for sure that they're ready to process an API request using this stream

Reply:
We could add something like a `connect` event that only fires when both sides are open

Reply:
Thinking about it, the HTTP server may be running some logic when the stream is opened and even though the stream module has fired the event, the HTTP server isn't ready for the HTTP client to send their second HTTP call. So it might actually be best to keep it this way, where the application layer logic signals that it's ready

Reply:
Ok, sounds good